p4-16/psa/PSA.mdk

```diff
@@ -120,6 +120,81 @@ Additional supported match_kind types
 
 # PSA Externs
 
+## Restrictions on where externs may be used { #sec-extern-restrictions }
+
+All instantiations in a P4~16~ program occur at compile time, and can
+be arranged in a tree structure we will call the instantiation tree.
+The root of the tree `T` represents the top level of the program.  Its
+children are the node for the package `PSA_Switch` described in
+[#sec-programmable-blocks], plus any externs instantiated at the top
+level of the program.  The children of the `PSA_Switch` node are the
+parsers and controls passed as parameters to the `PSA_Switch`
+instantiation.  If any of those parsers or controls instantiate other
+parsers, controls, and/or externs, create child nodes for them,
```

Comment:
"create child nodes for them" can be interpreted as asking the P4 programmer to create the nodes. We should says something like "the compiler creates nodes for them as children of the instantiating nodes" (or something to that effect).

Reply:
I am changing it to "the instantiation tree contains child nodes for them".